docs/ml-guide.md

```diff
@@ -61,6 +61,9 @@ To configure `netlib-java` / Breeze to use system optimised binaries, include
 project and read the [netlib-java](https://github.com/fommil/netlib-java) documentation for your
 platform's additional installation instructions.
 
+The most popular native BLAS such as [Intel MKL](https://software.intel.com/en-us/mkl), OpenBLAS, are based on multi-threading, which will conflict with Spark.
+To use multi-threading based native BLAS, you must set it to use single thread first (SPARK-21305).
```

Comment:
Perhaps "If using a native BLAS based on multi-threading, for best performance you must ..."